src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -462,8 +462,19 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send(const void *buf, MPI_Aint count, MPI
                     }
                 }
                 if (!fast_copy) {
-                    MPIR_Typerep_pack(buf, count, datatype, 0, host_buf, data_sz,
-                                      &actual_pack_bytes, MPIR_TYPEREP_FLAG_NONE);
+                    MPL_gpu_engine_type_t engine =
+                        MPIDI_OFI_gpu_get_send_engine_type(MPIR_CVAR_CH4_OFI_GPU_SEND_ENGINE_TYPE);
+                    if (dt_contig && engine != MPL_GPU_ENGINE_TYPE_LAST &&
+                        MPL_gpu_query_pointer_is_dev(send_buf, &attr)) {
```

Comment:
We use `attr.type == MPL_GPU_POINTER_DEV` in conditions above here in the code, but I think ZE defines what is "device" slightly differently? Just want to make sure I understand the difference.

Reply:
The difference being the call to `MPL_gpu_query_pointer_is_dev` here instead of checking `attr.type`.